amundsen_rds/models/table.py

```diff
@@ -0,0 +1,173 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+from sqlalchemy import (
+    BigInteger, Boolean, Column, ForeignKey, Integer, String, Text
+)
+from sqlalchemy.orm import relationship
+
+from amundsen_rds.models.base import (
+    KEY_LEN, NAME_LEN, PUBLISHED_TAG_LEN, URL_LEN, Base
+)
+
+
+class Table(Base):
+    """
+    Table model.
+    """
+    __tablename__ = 'table_metadata'
+
+    rk = Column(String(KEY_LEN), primary_key=True)
+    name = Column(String(NAME_LEN), nullable=False)
+    is_view = Column(Boolean, nullable=False)
+    schema_rk = Column(String(KEY_LEN), ForeignKey('schema_metadata.rk', ondelete='cascade'), nullable=False)
+    published_tag = Column(String(PUBLISHED_TAG_LEN), nullable=False)
+    publisher_last_updated_epoch_ms = Column(BigInteger)
+
+    application = relationship('Application', uselist=False, secondary='application_table')
+    schema = relationship('Schema', uselist=False)
+    columns = relationship('TableColumn', order_by='TableColumn.sort_order')
+    description = relationship('TableDescription', order_by='TableDescription.rk', uselist=False)
+    programmatic_descriptions = relationship('TableProgrammaticDescription', order_by='TableProgrammaticDescription.rk')
+    tags = relationship('Tag', order_by='Tag.rk', secondary='table_tag')
+    badges = relationship('Badge', order_by='Badge.rk', secondary='table_badge')
+    timestamp = relationship('TableTimestamp', uselist=False)
+    watermarks = relationship('TableWatermark', order_by='TableWatermark.rk')
+    source = relationship('TableSource', uselist=False)
+    usage = relationship('TableUsage', order_by='TableUsage.user_rk', backref='table')
```

Comment:
is uselist in orm.relationship default to true?

Reply:
Good question. The default value is None and ORM can automatically decide the value type on the direction of foreign key. Basically, uselist=False is only required in 1:1. In our models, for schema, uselist=False is actually optional.